provider/common/errors.go

```diff
@@ -80,3 +82,24 @@ var AuthorisationFailureStatusCodes = set.NewInts(
 	http.StatusForbidden,
 	http.StatusProxyAuthRequired,
 )
+
+// MaybeHandleCredentialError determines if a given error relates to an invalid credential.
+//  If it is, the credential is invalidated and the return bool is true.
+// Original error is returned untouched.
+func MaybeHandleCredentialError(isAuthError func(error) bool, err error, ctx context.ProviderCallContext) (error, bool) {
```

Comment:
If the original error is always returned untouched, it shouldn't return it.

Reply:
Ahga \o/ yes, historical reasons as some providers indeed do but this code would not...
The reason why the errors are returned untouched is so that they can be used used in one line of code in similar way to errors.Trace(), etc...